### PR TITLE
add and refactor existing tests according to ISL cost manipulation in DB

### PR DESCRIPTION
**Pull request by andriidovhan**
- refactor tests according to ISL cost manipulation in DB
- add test for checking that system takes into account the `time_unstable` field while creating a flow

### CODE REVIEW

File: services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/FlowCrudSpec.groovy [hunk 1]
```diff
@@ -817,6 +816,56 @@ class FlowCrudSpec extends HealthCheckSpecification {
         flows.each { flowHelper.deleteFlow(it) }
     }
 
+    def "System takes isl time_unstable info into account while creating a flow"() {
+        given: "Two active not neighboring switches with two possible paths at least"
+        def switchPair = topologyHelper.getAllNotNeighboringSwitchPairs().find {
+            it.paths.unique { it.first().portNo }.size() > 1
+        } ?: assumeTrue("No suiting switches found", false)
+
+        and: "Detect two available paths"
```
Comment (rtretyak) [inline services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/FlowCrudSpec.groovy]: I don't understand what I should do in this step

File: services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/FlowCrudSpec.groovy [hunk 2]
```diff
@@ -817,6 +816,56 @@ class FlowCrudSpec extends HealthCheckSpecification {
...
+        def firstPath = switchPair.paths.min { it.size() }
+        def secondPath = switchPair.paths.findAll { it != firstPath }.max { it.size() }
+        def altPaths = switchPair.paths.findAll {
+            it != firstPath && it.first().portNo != firstPath.first().portNo &&
+                    it != secondPath && it.first().portNo != secondPath.first().portNo
+        }.sort { it.size() }
+
+        and: "Make all alternative paths are unavailable (bring ports down on the srcSwitch)"
```
Comment (rtretyak) [inline services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/FlowCrudSpec.groovy]: `Make all alternative paths unavailable`

File: services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/FlowCrudSpec.groovy [hunk 3]
```diff
@@ -817,6 +816,56 @@ class FlowCrudSpec extends HealthCheckSpecification {
...
+        then: "Flow is created on the link from the second path"
+        PathHelper.convert(northbound.getFlowPath(flow.id)).first().portNo == secondPath.first().portNo
+
+        and: "Restore topology, delete the flow and reset costs"
+        broughtDownPorts.each { northbound.portUp(it.switchId, it.portNo) }
+        flowHelper.deleteFlow(flow.id)
+        Wrappers.wait(discoveryInterval + WAIT_OFFSET) {
+            northbound.getAllLinks().each { assert it.state != FAILED }
+        }
+        database.resetCosts()
```
Comment (rtretyak) [inline services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/FlowCrudSpec.groovy]: Is the ISL still in 'unstable' status at this point? Is there a way to get its current 'stability' status?

File: services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/links/LinkMaintenanceSpec.groovy [hunk 4]
```diff
@@ -11,19 +12,9 @@ import org.openkilda.functionaltests.helpers.Wrappers
...
 import org.openkilda.messaging.info.event.PathNode
 import org.openkilda.messaging.payload.flow.FlowState
-import org.openkilda.testing.Constants
-
-import org.springframework.beans.factory.annotation.Value
 
 class LinkMaintenanceSpec extends HealthCheckSpecification {
 
-    @Value('${isl.cost.when.under.maintenance}')
-    int islCostWhenUnderMaintenance
```
Comment (rtretyak) [inline services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/links/LinkMaintenanceSpec.groovy]: If this and `isl.cost.when.port.down` no longer exist, then it should be removed from all the templates and example properties

File: services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/FlowCrudSpec.groovy (hunk 1 above)
Comment (andriidovhan) [reply services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/FlowCrudSpec.groovy]: the step was renamed

File: services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/FlowCrudSpec.groovy (hunk 2 above)
Comment (andriidovhan) [reply services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/FlowCrudSpec.groovy]: thx

File: services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/FlowCrudSpec.groovy (hunk 3 above)
Comment (andriidovhan) [reply services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/FlowCrudSpec.groovy]: Actually there is no way to get status of ISL via API

File: services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/links/LinkMaintenanceSpec.groovy (hunk 4 above)
Comment (andriidovhan) [reply services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/links/LinkMaintenanceSpec.groovy]: in fact we still use this value during path computation.
we add this value to cost in real time and don't update cost for ISL in db

**Pull request merged**